### ISSUE TITLE
Update the webapp dynamically?

### ISSUE THREAD

**Issue by awesomebytes** [dddomodossola/remi#40]
I tried to make a little app where I check for some parameters of a running system and show them in a DropDown. I wanted to update this dropdown with the current parameters of the system (to choose one, and from there on, show also a dynamically generated table of the data to hopefully be edited). I made refresh button for trying to update them. I expected reloading the webpage to actually run the function "main" again, but I found out that does not happen. I used simple_app.py as my starting point.

So in short, is it possible or will it be possible to edit/add elements to the layout? I see you can set_text in some fields, but what about update dropdown contents or directly add new widgets on the fly?

Maybe it is just possible right now but I don't see how :P

**Comment by awesomebytes** [comment]
I actually found out that if I set:
https://github.com/dddomodossola/remi/blob/master/remi/server.py#L549

To True, on every reload of the page I'll get the main function executed again so I can get refreshed data.

Maybe this can be added as an option/parameter (this looks like you tried to optimize by only doing this once).

For adding dynamically widgets to the current webapp... I have no clue.

And, althought this may be like a different topic for another issue, can I get the parameters passed via GET/POST?

Thank you a lot for your efforts and your time.

**Comment by nzjrs** [comment]
I've recently been trying to move REMI towards the style of a normal gui toolkit (i.e. arbitary userdata attached to callbacks, multiple window support, etc).

At the moment you can pass userdata to main if you instantiate it via the Server object. This addresses half your problem as at least then you can pass in some outer state from your application and programatically *construct* the gui - see a minimal example [here](https://github.com/dddomodossola/remi/blob/master/examples/simple_app_multi.py#L8)

What do you want to achieve with the GET/POST parameters exactly? that's is an implementation detail. Do you want the value of the widget that changed?

**Comment by dddomodossola** [comment]
@awesomebytes the gui you construct is refreshed automatically by websockets. You can add elements and change its content at runtime, and they automatically updates. This is what is shown in widgets_overview_app.py, in particular in the callbacks. If you encounter problem, please show a peace of your code.
